src/meta/src/manager/background_deleter.rs

```diff
@@ -0,0 +1,176 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use itertools::Itertools;
+use risingwave_common::catalog::TableId;
+use risingwave_pb::catalog::{Sink, Source, Table};
+use tokio::sync::oneshot::Sender;
+use tokio::sync::{mpsc, oneshot};
+use tokio::task::JoinHandle;
+
+use super::SourceId;
+use crate::manager::MetaSrvEnv;
+use crate::model::{MetadataModel, MetadataModelResult};
+use crate::storage::MetaStore;
+use crate::stream::{GlobalStreamManagerRef, SourceManagerRef};
+use crate::MetaResult;
+
+pub type CatalogBackgroundDeleterRef = Arc<CatalogBackgroundDeleter>;
+
+macro_rules! create_catalog_deleted {
+    ($name:ident, $cf:ident) => {
+        paste::paste! {
+            #[derive(Debug)]
+            pub struct [<$name Deleted>](pub $name);
```

Comment:
Well, I think we don't have to persist all these catalogs. Since if fails, meta will trigger a recovery process and all these stream jobs will be cleaned.